src/Platform/Platform.php

```diff
@@ -0,0 +1,191 @@
+<?php
+
+namespace Utopia\Platform;
+
+use Utopia\App;
+use Utopia\CLI\CLI;
+use Exception;
+
+abstract class Platform
+{
+    protected array $services = [
+        'all' => [],
+        Service::TYPE_CLI => [],
+        Service::TYPE_HTTP => [],
+        Service::TYPE_GRAPHQL => []
+    ];
+
+    protected CLI $cli;
+
+    /**
+     * Initialize Application
+     *
+     * @return void
+     */
+    public function init(string $type = null): void
+    {
+        switch ($type) {
+            case Service::TYPE_HTTP:
+                $this->initHttp();
+                break;
+            case Service::TYPE_CLI:
+                $this->initCLI();
+                break;
+            case Service::TYPE_GRAPHQL:
+                $this->initGraphQL();
+                break;
+            case 'all':
+            default:
```

Comment:
Seems like this is not relevant. We can throw an exception and have no default value in the method signature.

Reply:
also removed case 'all' as it seems we cannot initialize CLI and HTTP together